tokio/src/runtime/builder.rs

```diff
@@ -204,70 +222,129 @@ impl Builder {
         self
     }
 
+    /// Set the `Clock` instance that will be used by the runtime.
+    pub fn clock(&mut self, clock: Clock) -> &mut Self {
+        self.clock = clock;
+        self
+    }
+
     /// Create the configured `Runtime`.
     ///
     /// The returned `ThreadPool` instance is ready to spawn tasks.
     ///
     /// # Examples
     ///
     /// ```
-    /// # use tokio::runtime::Builder;
-    /// # pub fn main() {
-    /// let runtime = Builder::new().build().unwrap();
-    /// // ... call runtime.run(...)
-    /// # let _ = runtime;
-    /// # }
+    /// use tokio::runtime::Builder;
+    ///
+    /// let mut rt = Builder::new().build().unwrap();
+    ///
+    /// rt.block_on(async {
+    ///     println!("Hello from the Tokio runtime");
+    /// });
     /// ```
     pub fn build(&mut self) -> io::Result<Runtime> {
-        let mut reactor_handles = Vec::new();
+        if self.current_thread {
+            self.build_current_thread()
+        } else {
+            self.build_threadpool()
+        }
+    }
+
+    fn build_current_thread(&mut self) -> io::Result<Runtime> {
+        // Create network driver
+        let net = Reactor::new()?;
+        let net_handles = vec![net.handle()];
+
+        let timer = Timer::new_with_clock(net, self.clock.clone());
+        let timer_handles = vec![timer.handle()];
+
+        // And now put a single-threaded executor on top of the timer. When
+        // there are no futures ready to do something, it'll let the timer or
+        // the reactor to generate some new stimuli for the futures to continue
+        // in their life.
+        let executor = CurrentThread::new(timer);
+
+        // Blocking pool
+        let blocking_pool = PoolWaiter::from(Pool::default());
```

Comment:
I forget exactly what `Pool::default()` is, but we should make sure that for `current_thread` no blocking threads should be spawned by default.

Reply:
No threads are spawned by default.